### PR TITLE
[Robot] added tests for validation rules on SD

### PR DESCRIPTION
**Pull request by rmonica08**
Added test to
1. Validate contact and account lookup to the same household on new service delivery dialog
2. Validate Service and program engagement lookup to the same program on new service delivery quick action
3. Validate Contact matches the contact in program engagement on new service delivery quick action
4.  Create a new service delivery using quick action from service record

W-042516

# Critical Changes

# Changes

# Issues Closed

# New Metadata

# Deleted Metadata

# Definition of Done
  Refer to [Asteroids DoD document](https://salesforce.quip.com/iq2mAy4i62oM) to see any additional details for the items below
- [ ] Place holder data is incorporated (Refer to [Case Management Sample Data document](https://quip.com/cbFcAPJF8t0z))
- [ ] Base axe-core JEST test included for any new LWC (No critical violations)
- [ ] CRUD/FLS is enforced in Apex
- [ ] Permission sets are updated to account for CRUD/FLS for new object metadata
- [ ] All modified classes are unit tested (Apex) at 100% coverage
- [ ] UX approval or UX not necessary
- [ ] PR contains draft release notes
- [ ] Attach work item to the pull request with [Lurch](https://salesforce.quip.com/50ZRA5LEWVzH) `**Lurch:attach W-000000` or `**Lurch:add`
- [ ] All **acceptance criteria** have been met
    - [ ] Developer
    - [ ] Code Reviewer
    - [ ] QA
- [ ] QE story level testing completed

### CODE REVIEW

File: robot/pmm/tests/browser/ProgramEngagement/quick_actions_sd.robot [hunk 1]
```diff
@@ -1,50 +1,65 @@
...
-     Load Page Object            NewServiceDelivery   ${ns}ServiceDelivery__c
-     verify current page title   Create New Service Delivery
-     populate modal form         Client=&{contact}[FirstName] &{contact}[LastName]
-     ...                         Service=&{service}[Name]
-     ...                         Quantity=${quantity}
-     Click Modal button          Save
+     Go To Page                     Details                        ProgramEngagement__c      object_id=${program_engagement}[Id]
+     Verify Details                 Program Engagement Name        contains                       ${contact}[FirstName] ${contact}[LastName] ${today}: ${program}[Name]
+     Click Quick Action Button      Create New Service Delivery
+     Verify Current Page Title      Create New Service Delivery
```
Comment (skristem) [inline robot/pmm/tests/browser/ProgramEngagement/quick_actions_sd.robot:42]: can this verify step be added as part of click quick action button?

File: robot/pmm/tests/browser/ProgramEngagement/quick_actions_sd.robot [hunk 2]
```diff
@@ -1,50 +1,65 @@
...
+     Click Modal Button             Save
      Wait Until Modal Is Closed
-     verify page header          Program Engagement
-     Load Related List           Service Deliveries
-     click new related record link  &{contact}[FirstName] &{contact}[LastName] ${today}: &{service}[Name]
-     verify details             Service Delivery Name   contains                            &{contact}[FirstName] &{contact}[LastName] ${today}: &{service}[Name]
-     ${servicedelivery_id} =    Save Current Record ID For Deletion     ${ns}ServiceDelivery__c
-     ${service_id} =            Save Current Record ID For Deletion     ${ns}Service__c
-     ${programengagement_id} =  Save Current Record ID For Deletion     ${ns}ProgramEngagement__c
+     Verify Page Header             Program Engagement
```
Comment (skristem) [inline robot/pmm/tests/browser/ProgramEngagement/quick_actions_sd.robot:48]: should we instead be using current page should be? can verification of page header could be added as part of that?

File: robot/pmm/tests/browser/ProgramEngagement/quick_actions_sd.robot [hunk 3]
```diff
@@ -1,50 +1,65 @@
...
+     Save Current Record ID For Deletion     ${ns}ServiceDelivery__c
+
+Validate contact on quick action dialog
+     [Documentation]                This test loads the program engagement record, clicks on the new service delivery quick action and
+     ...                            validates an error message is displayed when a contact not listed on program engagement is entered
+     [tags]                         W-042516  feature:Service Delivery
+     Go To Page                     Details                        ProgramEngagement__c     object_id=${program_engagement}[Id]
+     Verify Details                 Program Engagement Name        contains                      ${contact}[FirstName] ${contact}[LastName] ${today}: ${program}[Name]
+     Click Quick Action Button      Create New Service Delivery
+     Verify Current Page Title      Create New Service Delivery
```
Comment (skristem) [inline robot/pmm/tests/browser/ProgramEngagement/quick_actions_sd.robot:61]: same comment for this one as well

File: robot/pmm/tests/browser/Service/quickactions_service.robot [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+     Go To Page                     Details                        Service__c           object_id=${service}[Id]
+     Verify Details                 Service Name                   contains                  ${service}[Name]
+     Click Quick Action Button      Create New Service Delivery
+     Verify Current Page Title      Create New Service Delivery
+     Populate Modal Form            Client=${contact}[FirstName] ${contact}[LastName]
+     ...                            Program Engagement=${program_engagement}[Name]
+     ...                            Quantity=${quantity}
+     Click Modal Button             Save
+     Wait Until Modal Is Closed
+     Verify Page Header             Service
```
Comment (skristem) [inline robot/pmm/tests/browser/Service/quickactions_service.robot:52]: same current page should be comment for this one as well

File: robot/pmm/tests/browser/Service/quickactions_service.robot [hunk 5]
```diff
@@ -0,0 +1,70 @@
...
+*** Test Cases ***
+Create a new service delivery on service using quick action
+     [Documentation]                This test loads the service record, clicks on the new service delivery quick action and creates
+     ...                            new service delivery record.
+     [tags]                         W-042516  feature:Service Delivery
+     Go To PMM App
+     Go To Page                     Details                        Service__c           object_id=${service}[Id]
+     Verify Details                 Service Name                   contains                  ${service}[Name]
+     Click Quick Action Button      Create New Service Delivery
+     Verify Current Page Title      Create New Service Delivery
```
Comment (skristem) [inline robot/pmm/tests/browser/Service/quickactions_service.robot:46]: same here

File: robot/pmm/tests/browser/Service/quickactions_service.robot [hunk 6]
```diff
@@ -0,0 +1,70 @@
...
+     Save Current Record ID For Deletion     ${ns}ServiceDelivery__c
+
+Validate service and program engagement lookup to same program
+     [Documentation]                This test loads the program engagement record, clicks on the new service delivery quick action and
+     ...                            validates an error message is displayed when service and program engagement do not lookup to the same program
+     [tags]                         W-042516  feature:Service Delivery
+     Go To Page                     Details                        Service__c           object_id=${service}[Id]
+     Verify Details                 Service Name                   contains                  ${service}[Name]
+     Click Quick Action Button      Create New Service Delivery
+     Verify Current Page Title      Create New Service Delivery
```
Comment (skristem) [inline robot/pmm/tests/browser/Service/quickactions_service.robot:65]: same here

File: robot/pmm/tests/browser/ServiceDeliveries/new_service_delivery.robot [hunk 7]
```diff
@@ -18,67 +18,72 @@ Setup Test Data
...
-    verify details                         Quantity                                contains             ${quantity}
-    verify details                         Service                                 contains             &{service}[Name]
-    verify details                         Program Engagement                      contains             &{program_engagement}[Name]
-    page should not contain                ${service_delivery_name}
-    verify page contains related list      Files
-    ${service_delivery_id} =               Save Current Record ID For Deletion     ${ns}ServiceDelivery__c
-    ${service_id} =                        Save Current Record ID For Deletion     ${ns}Service__c
-    ${program_engagement_id} =             Save Current Record ID For Deletion     ${ns}ProgramEngagement__c
-
+    Verify Page Header                     Service Delivery
```
Comment (skristem) [inline robot/pmm/tests/browser/ServiceDeliveries/new_service_delivery.robot:49]: this one as well

File: robot/pmm/tests/browser/ServiceDeliveries/new_service_delivery.robot [hunk 8]
```diff
@@ -18,67 +18,72 @@ Setup Test Data
...
+    Verify Details                         Quantity                                contains             ${quantity}
+    Verify Details                         Service                                 contains             ${service}[Name]
+    Verify Details                         Program Engagement                      contains             ${program_engagement}[Name]
+    Page Should Not Contain                ${service_delivery_name}
+    Verify Page Contains Related List      Files
+    Save Current Record ID For Deletion     ${ns}ServiceDelivery__c
 
 Create a Service Delivery via UI with Auto Name Override
     [Documentation]                        This test creates Service Delivery record and verifies that the Service Delivery record
```
Comment (skristem) [inline robot/pmm/tests/browser/ServiceDeliveries/new_service_delivery.robot:58]: documentation for this test is exactly same as above one, please correct

File: robot/pmm/tests/browser/ServiceDeliveries/new_service_delivery.robot [hunk 9]
```diff
@@ -18,67 +18,72 @@ Setup Test Data
...
+    Click Modal Button                     Save
     Wait Until Modal Is Closed
-    verify page header                     Service Delivery
-    verify details                         Service Delivery Name                   contains             ${service_delivery_name}
-    verify page contains related list      Files
-    ${service_delivery_id} =               Save Current Record ID For Deletion     ${ns}ServiceDelivery__c
-    ${service_id} =                        Save Current Record ID For Deletion     ${ns}Service__c
-    ${program_engagement_id} =             Save Current Record ID For Deletion     ${ns}ProgramEngagement__c
\ No newline at end of file
+    Verify Page Header                     Service Delivery
```
Comment (skristem) [inline robot/pmm/tests/browser/ServiceDeliveries/new_service_delivery.robot:72]: same comment

Comment (rmonica08) [reply robot/pmm/tests/browser/ServiceDeliveries/new_service_delivery.robot]: Fixed

File: robot/pmm/tests/browser/ServiceDeliveries/new_service_delivery.robot (hunk 8 above)
Comment (rmonica08) [reply robot/pmm/tests/browser/ServiceDeliveries/new_service_delivery.robot]: Fixed

File: robot/pmm/tests/browser/ServiceDeliveries/new_service_delivery.robot (hunk 7 above)
Comment (rmonica08) [reply robot/pmm/tests/browser/ServiceDeliveries/new_service_delivery.robot]: used Current Page Should Be keyword

File: robot/pmm/tests/browser/Service/quickactions_service.robot (hunk 5 above)
Comment (rmonica08) [reply robot/pmm/tests/browser/Service/quickactions_service.robot]: Combined both into Click Quick Action Button keyword

File: robot/pmm/tests/browser/Service/quickactions_service.robot (hunk 4 above)
Comment (rmonica08) [reply robot/pmm/tests/browser/Service/quickactions_service.robot]: Fixed

File: robot/pmm/tests/browser/ProgramEngagement/quick_actions_sd.robot (hunk 1 above)
Comment (rmonica08) [reply robot/pmm/tests/browser/ProgramEngagement/quick_actions_sd.robot]: Fixed

File: robot/pmm/tests/browser/ProgramEngagement/quick_actions_sd.robot (hunk 2 above)
Comment (rmonica08) [reply robot/pmm/tests/browser/ProgramEngagement/quick_actions_sd.robot]: Fixed, also removed 'Validate contact on quick action dialog' test as the contact is now autopopulated when the quick action button is clicked, so there is no need to validate contact on quickaction

File: robot/pmm/tests/browser/ServiceDeliveries/new_service_delivery.robot [hunk 10]
```diff
@@ -18,67 +18,74 @@ Setup Test Data
...
     Wait Until Modal Is Closed
-    verify page header                     Service Delivery
-    verify details                         Service Delivery Name                   contains             ${service_delivery_name}
-    verify page contains related list      Files
-    ${service_delivery_id} =               Save Current Record ID For Deletion     ${ns}ServiceDelivery__c
-    ${service_id} =                        Save Current Record ID For Deletion     ${ns}Service__c
-    ${program_engagement_id} =             Save Current Record ID For Deletion     ${ns}ProgramEngagement__c
\ No newline at end of file
+    Current Page Should Be                 Details                                 ServiceDelivery__c
+   # Verify Page Header                     Service Delivery
```
Comment (skristem) [inline robot/pmm/tests/browser/ServiceDeliveries/new_service_delivery.robot:74]: remove this commented out line

Comment (rmonica08) [reply robot/pmm/tests/browser/ServiceDeliveries/new_service_delivery.robot]: Removed and also updated the is_current_page keyword

**Pull request merged**